cvmfs/magic_xattr.cc

```diff
@@ -222,28 +227,75 @@ std::string CatalogCountersMagicXattr::GetValue() {
   return res;
 }
 
-bool ChunkListMagicXattr::PrepareValueFenced() {
-  chunk_list_ = "hash,offset,size\n";
+bool ChunkListMagicXattr::PrepareValueFenced(int32_t attr_page_num) {
+  chunk_list_ = "ERROR: No chunks! ";
   if (!dirent_->IsRegular()) {
+    chunk_list_ += "Not a regular file\n";
     return false;
   }
+
   if (dirent_->IsChunkedFile()) {
     FileChunkList chunks;
     if (!xattr_mgr_->mount_point()->catalog_mgr()
                      ->ListFileChunks(path_, dirent_->hash_algorithm(), &chunks)
         || chunks.IsEmpty())
     {
+      chunk_list_ += "File marked as chunked, but no chunks found\n";
       LogCvmfs(kLogCvmfs, kLogDebug | kLogSyslogErr, "file %s is marked as "
                 "'chunked', but no chunks found.", path_.c_str());
       return false;
     } else {
-      for (size_t i = 0; i < chunks.size(); ++i) {
+      const size_t kB_limit = 64 * 1024;
```

Comment:
Let's add a good safety margin for characters that may be added elsewhere. I'd suggest 40000 characters payload max.